drivers/adc/Kconfig.mcux

```diff
@@ -53,6 +53,45 @@ config ADC_MCUX_ADC16_VREF_ALTERNATE
 
 endchoice
 
+config ADC16_ENABLE_EDMA
+	bool "ENABLE EDMA for adc driver"
+	depends on HAS_MCUX_ADC16 && HAS_MCUX_EDMA
+	help
+	  Enable the MCUX ADC16 driver.
+
+if ADC16_ENABLE_EDMA
+
+config ADC16_INIT_PRIORITY
+	int "Init priority"
+	default 70
+	help
+	  Device driver initialization priority.
+
+config ADC_EDMA_CHANNEL
+	int "ADC DMA Channel select"
+	depends on HAS_MCUX_EDMA
+	default 0
+	help
+	  ADC DMA Channel select (0-4) PIT triggerred
+
+config ADC_HW_TRIGGER
+	bool "ADC HW TRIGGER"
+	default y
+	help
+	  Support HW Trigger ADC
+
+if ADC_HW_TRIGGER
+
+config ADC_HW_TRIGGER_CHANNEL
+	int "HW trigger ADC channel"
+	default 4
+	help
+	  hw channel id as trigger source default PIT0
```

Comment:
Could we move this to the DTS? Or how would you configure different DMA channels/HW triggers per ADC16 instance?

Reply:
this is a SOC setting, set to SIM->SOPT7 bit8-11, different SOC has slightly different for each possible trigger source. if move to DTS, it is not in ADC controller, but in SIM module. see below diagram
![image](https://user-images.githubusercontent.com/800198/99765451-0eb87800-2b3a-11eb-84c7-81467b4a9c0b.png)
it is not to config the DMA channels but the ADC16 trigger source, DMA channel is fixed for ADC in given SOC, see below in ADC DTS dmas. 
		adc0: adc@4003b000{
	             ... 
			dmas = <&edma0 0 40>;
			dma-names = "adc0";
                    ...
               		};

Reply:
How would you configure ADC0 to use PIT trigger 0, and ADC1 to use PIT trigger 1?

Reply:
this is actually controlled by user, I add a test application to example how to use pit0 to trigger by configuring the ADC_HW_TRIGGER_CHANNEL to 4.  this is set in the driver code  mcux_adc16_channel_setup by hardcode to adc0 with this marco.  
let me propose a way to use DTS setting to enable.

Reply:
@henrikbrixandersen  add to DTS hw_trigger_src , please check.

```
&adc0 {
	clk_source = <0>;
	hw_trigger_src = <4>;
	nxp,continues_convert;
	nxp,high_speed;
};
```